test/cli/all_formats_convertible_test.go

```diff
@@ -0,0 +1,81 @@
+package cli
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+type conversion struct {
+	To   string
+	From string
+}
+
+func TestConvertCmdFlags(t *testing.T) {
+	commonAssertions := []traitAssertion{
+		func(tb testing.TB, stdout, _ string, _ int) {
+			tb.Helper()
+			if len(stdout) < 1000 {
+				tb.Errorf("there may not be any report output (len=%d)", len(stdout))
+			}
+		},
+		assertSuccessfulReturnCode,
+	}
+
+	tests := []struct {
+		name        string
+		conversions []conversion
+		env         map[string]string
+		assertions  []traitAssertion
+	}{
+		{
+			name: "syft-format convertable to spdx-json",
+			conversions: []conversion{
+				{To: "syft-json", From: "spdx-json"},
+				{To: "syft-json", From: "cyclonedx-json"},
+				{To: "spdx-json", From: "syft-json"},
+				{To: "spdx-json", From: "cyclonedx-json"},
+				{To: "cyclonedx-json", From: "syft-json"},
+				{To: "cyclonedx-json", From: "spdx-json"},
+			},
+			assertions: commonAssertions,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			for _, c := range test.conversions {
+				sbomArgs := []string{"dir:./test-fixtures/image-pkg-coverage", "-o", c.From}
+				cmd, stdout, stderr := runSyft(t, test.env, sbomArgs...)
+				if cmd.ProcessState.ExitCode() != 0 {
+					t.Fatalf("failure executing syft creating an sbom")
+					t.Log("STDOUT:\n", stdout)
+					t.Log("STDERR:\n", stderr)
+					t.Log("COMMAND:", strings.Join(cmd.Args, " "))
+					return
+				}
+
+				f, err := os.CreateTemp("", "temp_sbom")
+				if err != nil {
+					t.Fatalf("could not create temp sbom file for convert: %s", err)
+				}
+				defer os.Remove(f.Name()) // clean up temp file
+
+				if _, err := f.Write([]byte(stdout)); err != nil {
+					t.Fatalf("could not write temp sbom for convert: %s", err)
+				}
```

Comment:
nit:
```suggestion
				tempDir := t.TempDir()
				sbomFile := filepath.Join(tempDir, "sbom.json")
				require.NoError(t, os.WriteFile(sbomFile, stdout, 0666))
```
+ a few other `f.Name()` calls